[jason] iP

Jasonzhou97 opened the pull request:
# Lebum Chatbot - The GOAT Task Manager 👑

> "I promise to handle your tasks like I handle the fourth quarter" - Lebum James

Lebum frees your mind from task management, just like how LeGM frees himself from his noob teammates. It's:

* Built **DIFFERENT**
* More clutch than Game 6 Klay

All you need to do is:
1. Download it from [here](https://github.com/jasonzhou97/ip)
2. Double-click to start the greatness
3. Enter commands like `todo`, `deadline`, or `event`
4. Watch Lebum cook 🔥

Development Progress:
- [x] Basic command handling
- [x] JavaFX GUI implementation
- [x] Task storage system
- [ ] Taco Tuesday mode

Here's how it happens:

```java
public class Main extends Application {
    @Override
    public void start(Stage stage) {
        // Taking my talents to JavaFX
        Application.launch(Main.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jasonzhou97 committed:
Added greeting and exiting messages

Jasonzhou97 committed:
added level-1 functions

Jasonzhou97 committed:
added tasks and list functions

Jasonzhou97 committed:
added marking and unmarking

Jasonzhou97 committed:
added subclasses and their functions

Jasonzhou97 committed:
added automated testing

Jasonzhou97 committed:
added exception catching and updated automated testing

Jasonzhou97 committed:
update testing input and expected result

Jasonzhou97 committed:
add delete functions

Jasonzhou97 committed:
add more javadoc comments

Jasonzhou97 committed:
Update Duke and TaskList implementation

Jasonzhou97 committed:
added saving and loading from txt file

Jasonzhou97 committed:
Merge branch-Level-7

Jasonzhou97 committed:
added datetime object features

Jasonzhou97 committed:
added command class and ui calss

Jasonzhou97 committed:
Add OOP classes for commands and parser and UI

Jasonzhou97 committed:
Merge A-MoreOOP

Jasonzhou97 committed:
split files into packages

Jasonzhou97 committed:
Merge branch 'add-gradle-support'
Merge add-gradle-support branch to set up Gradle build

Jasonzhou97 committed:
added junit test cases

Jasonzhou97 committed:
Add and refine JUnit test cases

Jasonzhou97 committed:
Fix bugs relating to creating directory upon first using

Jasonzhou97 committed:
Add more javadoc comments

Jasonzhou97 committed:
Tweaked some code to fit java coding standard

Jasonzhou97 committed:
Add finding keyword functionalities

Jasonzhou97 committed:
Add basic GUI

Jasonzhou97 committed:
Edit main class of launcher and duke methods

Jasonzhou97 committed:
Add list GUI functionalities

Jasonzhou97 committed:
Refine GUI commands

Windofbitter commented on src/main/java/duke/main/TaskList.java:
```diff
@@ -28,7 +28,9 @@ public String addTask(Task task) {
         num_of_tasks += 1;
```
Maybe we can try to stick to CamelCase to follow consistency in naming

Windofbitter commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -6,16 +6,21 @@ public class MarkCommand extends Command{
             Task curTask = tasks.array().get(index);
+            response = "You have marked " + curTask.getTitle() + " as done.";
             curTask.markDone();
             storage.saveToFile(tasks);
```
I love the way you name all the methods, it is very clear and of high quality

Windofbitter commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -26,6 +26,7 @@ public void execute(TaskList tasks, Storage storage, Ui ui) throws DukeException
                 for (String p: this.parts){
```
Maybe we can try to leave 4 spaces after such statements to follow the coding standard proposed by the course

Jasonzhou97 committed:
Add GUI messages functions and rename classes

Jasonzhou97 committed:
Add more  GUI interactions and input handling.

Stickiem commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,45 @@
+    @Override
+    public void execute(TaskList tasks, Storage storage, Ui ui) throws DukeException {
+        try {
+            boolean found = false;
```
Should this be named "isFound" instead?

Stickiem commented on src/test/java/duke/main/ParserTest.java:
```diff
@@ -0,0 +1,26 @@
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+import duke.command.*;
```
Should the imported classes be listed explicitly?

Stickiem commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,49 @@
+        String[] parts = input.split(" ",2);
+        String command = parts[0].toLowerCase();
+
+        switch(command){
+            case "list":
+                return new ListCommand();
+
+            case "delete":
+                return new DeleteCommand(parts);
+
+            case "todo":
+                return new AddToDoCommand(input);
+
+
+            case "deadline":
+                return new AddDeadlineCommand(input);
+
+            case "event":
+                return new AddEventCommand(input);
+
+            case "mark":
+                return new MarkCommand(parts);
+
+            case "unmark":
+                return new UnmarkCommand(parts);
+
+            case "bye":
+                return new ExitCommand();
+
+            case "find":
+                return new FindCommand(parts);
+            default:
+                throw new DukeException("OOPS I have no idea what that means :(");
+
+
+        }
```
Should there be indentation for the case clauses?

Stickiem commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,62 @@
+package duke.command;
+import duke.exception.DukeException;
+import duke.main.Storage;
+import duke.main.TaskList;
+import duke.main.Ui;
+import duke.task.Deadline;
```
Should the ordering of the import be consistent?

Stickiem reviewed (commented):
Looks good.
Just some coding standards related changes requested.

Jasonzhou97 committed:
Edit lebum bot and user avatar images

Jasonzhou97 committed:
Customise bot greetings

Jasonzhou97 committed:
Add assertions to validate method parameters and class invariants

The code accepts parameters and modifies object state without verifying
their validity. State inconsistencies can propagate silently through
the application.

Invalid inputs and state mutations can cause bugs that are hard to
track down, especially in larger code bases like later stage of ip and tp.

I've added assertions to validate:
* Method parameters before processing eg commands
* Object state after modifications

Assertions serve as self-documenting code and help catch bugs early
in development. They make the code's assumptions explicit and help
other developers understand the code's expectations.

Note: Assertions are disabled by default in production builds.

Jasonzhou97 committed:
Enhance AddDeadlineCommand with validation and streams

- Extract validation logic to separate validateInput() method
- Add constants for error messages
- Use streams for task filtering and manipulation
- Improve exception handling specificity
- Add trim() for input sanitization

Previous code used imperative loops and inline validation.
New changes improve maintainability and readability.

Jasonzhou97 committed:
Improve java code standards and documents

Jasonzhou97 committed:
Add massop extension functions like mass delete

Jasonzhou97 committed:
Merge pull request #1 from Jasonzhou97/branch-A-Assertions

Add assertions to validate method parameters and class invariants

Jasonzhou97 committed:
Merge pull request #2 from Jasonzhou97/A-Extension

A Extension

Jasonzhou97 committed:
Add mass mark and mass unmark functions

Jasonzhou97 committed:
Merge remote changes and add mass mark/unmark functionality

Jasonzhou97 committed:
Fix bugs relating to printing on same line when user adds tasks

Jasonzhou97 committed:
Improve styling for chat interface

Jasonzhou97 committed:
Resolve merge conflicts with branch-A-CodeQuality

Jasonzhou97 committed:
Improve styling for chat interface

Jasonzhou97 committed:
Add personality welcome and bye messages

Jasonzhou97 committed:
Add more error handling for deadline class

Jasonzhou97 committed:
Fix coding quality and add new greet command

Jasonzhou97 committed:
Add greeting commands

Jasonzhou97 committed:
Update user guide

Jasonzhou97 committed:
Fix coding qualities issues

Jasonzhou97 committed:
Update readme.md

Jasonzhou97 committed:
Fix image route issue in readme.md

Jasonzhou97 committed:
Add more error handling

Jasonzhou97 committed:
Improve bye command

Jasonzhou97 committed:
Improve checkstyle

Jasonzhou97 committed:
Add more junit tests

Jasonzhou97 committed:
Edit styling

Jasonzhou97 committed:
Improve error catching

Jasonzhou97 committed:
Fix io error

Jasonzhou97 committed:
Fix unmark bug

Jasonzhou97 committed:
Rename duke files

Jasonzhou97 committed:
Edit build gradle file
